store/sqlstore/group_supplier.go

```diff
@@ -0,0 +1,864 @@
+	result := store.NewSupplierResult()
+
+	if err := groupSyncable.IsValid(); err != nil {
+		result.Err = err
+		return result
+	}
+
+	// Reset values that shouldn't be updatable by parameter
+	groupSyncable.DeleteAt = 0
+	groupSyncable.CreateAt = model.GetMillis()
+	groupSyncable.UpdateAt = groupSyncable.CreateAt
+
+	var err error
+
+	switch groupSyncable.Type {
+	case model.GroupSyncableTypeTeam:
+		teamResult := <-s.Team().Get(groupSyncable.SyncableId)
+		if teamResult.Err != nil {
+			result.Err = teamResult.Err
+			return result
+		}
+
+		err = s.GetMaster().Insert(groupSyncableToGroupTeam(groupSyncable))
+	case model.GroupSyncableTypeChannel:
+		channelResult := <-s.Channel().Get(groupSyncable.SyncableId, false)
+		if channelResult.Err != nil {
+			result.Err = channelResult.Err
+			return result
+		}
+
+		err = s.GetMaster().Insert(groupSyncableToGroupChannel(groupSyncable))
+	default:
+		result.Err = model.NewAppError("SqlGroupStore.GroupCreateGroupSyncable", "model.group.type.app_error", nil, "group_id="+groupSyncable.GroupId+", syncable_id="+groupSyncable.SyncableId+", "+err.Error(), http.StatusInternalServerError)
+		return result
+	}
+	if err != nil {
+		if err == sql.ErrNoRows {
+			result.Err = model.NewAppError("SqlGroupStore.GroupCreateGroupSyncable", "store.sql_group.no_rows_affected", nil, "group_id="+groupSyncable.GroupId+", syncable_id="+groupSyncable.SyncableId, http.StatusInternalServerError)
+		}
+		result.Err = model.NewAppError("SqlGroupStore.GroupCreateGroupSyncable", "store.insert_error", nil, "group_id="+groupSyncable.GroupId+", syncable_id="+groupSyncable.SyncableId+", "+err.Error(), http.StatusInternalServerError)
+		return result
+	}
+
+	result.Data = groupSyncable
+	return result
+}
+
+func (s *SqlSupplier) GroupGetGroupSyncable(ctx context.Context, groupID string, syncableID string, syncableType model.GroupSyncableType, hints ...store.LayeredStoreHint) *store.LayeredStoreSupplierResult {
+	result := store.NewSupplierResult()
+
+	groupSyncable, err := s.getGroupSyncable(groupID, syncableID, syncableType)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			result.Err = model.NewAppError("SqlGroupStore.GroupGetGroupSyncable", "store.sql_group.no_rows", nil, err.Error(), http.StatusNotFound)
+		} else {
+			result.Err = model.NewAppError("SqlGroupStore.GroupGetGroupSyncable", "store.select_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+		return result
+	}
+
+	result.Data = groupSyncable
+
+	return result
+}
+
+func (s *SqlSupplier) getGroupSyncable(groupID string, syncableID string, syncableType model.GroupSyncableType) (*model.GroupSyncable, error) {
+	var err error
+	var result interface{}
+
+	switch syncableType {
+	case model.GroupSyncableTypeTeam:
+		result, err = s.GetMaster().Get(groupTeam{}, groupID, syncableID)
+	case model.GroupSyncableTypeChannel:
+		result, err = s.GetMaster().Get(groupChannel{}, groupID, syncableID)
+	default:
```

Comment:
nit: omit empty `default` if we're not going to do anything anyway? Or should we be error checking here?